### PR TITLE
Fixed compile issue re: NO_BLADE_NO_GEST_ONOFF and usage edits.

### PR DESCRIPTION
**Pull request by NoSloppy**
Someone ate my PR....anyway,
In case NO_BLADE_NO_GEST_ONOFF is used without BLADE_DETECT_PIN defined.

### CODE REVIEW

File: props/saber_BC_buttons.h [hunk 1]
```diff
@@ -1039,6 +1040,7 @@ class SaberBCButtons : public virtual PropBase {
   bool battle_mode_ = false;
   bool max_vol_reached_ = false;
   bool min_vol_reached_ = false;
+  bool blade_detected_ = true;
```
Comment (profezzorn) [inline props/saber_BC_buttons.h:1043]: This creates two variables with the same name which is almost never a good idea.
Unless you have a terribly good reason for this, please solve it some other way.

**Comment by NoSloppy** [comment]
Do you think it's okay that I leave it up to the end user based on the
instructions up top that a blade detect pin needs to be defined to use it?
I'm fine with that. I just happened to compile with only one of them
defined and got errors and was trying to keep it foolproof. Now I see where
it could be a problem this way. I went this route to avoid adding a whole
bunch of #ifndef blade detect pin lines in the middle of all the code
so.... Just let it go?

**Comment by profezzorn** [comment]
I'm thinking the easy thing to do would be something like:

#if defined(NO_BLADE_NO_GEST_ONOFF) && !defined(BLADE_DETECT_PIN)
#error You need blade detect (BLADE_DETECT_PIN) to use
NO_BLADE_NO_GEST_ONOFF
#endif

**Comment by NoSloppy** [comment]
Oh ok that's great, thank you. Now I know.

**Pull request merged**